### ISSUE TITLE
Allow for provider state handler timeout to be overridden

### ISSUE THREAD

**Issue by nsfrias** [pact-foundation/pact-go#277]
## Software versions

* **OS**: e.g. Mac OSX 13.0.1
* **Consumer Pact library**: Pact Go CLI v2.0.0-beta.17
* **Provider Pact library**: Pact Go CLI v2.0.0-beta.17
* **libpact_ffi**:  0.3.2 
* **Golang Version**: `1.18.3`

## Expected behaviour

Be able to set a timeout for provider state handler for a specific provider test.

## Actual behaviour

Currently there is no way to override the provider state handler timeout for a provider test, which means that if a state handler takes a long time to execute the provider test fails.

```
Err(Provider state failed: (interaction_id: ...) Invalid response: error sending request for url (http://localhost:50159/__setup): operation timed out)
```

**Comment by mefellows** [comment]
Just pushed a branch up with a small change that should support setting this timeout (and also disabling SSL verification if desired)

**Comment by mefellows** [comment]
Let me know if the PR fixes your issue. Build is green and good to go (I think it's worth merging regardless, but would appreciate a second pair of 👀 )

**Comment by nsfrias** [comment]
Works like a charm.

**Comment by mefellows** [comment]
Lovely! Thanks for confirming. Merging and releasing...

**Comment by mefellows** [comment]
Released v2.0.0-beta.18.
